### PR TITLE
[FIX] race condition with Owl control panel in mobile

### PR DESCRIPTION
**Pull request by aab-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by ged-odoo — APPROVED** [review]

File: addons/web/static/tests/fields/relational_fields/field_one2many_tests.js [hunk 1]
```diff
@@ -6885,7 +6910,11 @@ QUnit.module('fields', {}, function () {
             });
 
             await testUtils.dom.click(form.$('.o_field_x2many_list_row_add a'));
-
+            // use of owlCompatibilityNextTick because we have a boolean field (owl) inside the
+            // x2many, so an update of the x2many requires to wait for 2 animation frame: one
```
Comment (Polymorphe57) [inline addons/web/static/tests/fields/relational_fields/field_one2many_tests.js:6914]: frames

File: addons/web/static/tests/fields/relational_fields/field_one2many_tests.js [hunk 2]
```diff
@@ -4541,6 +4552,12 @@ QUnit.module('fields', {}, function () {
 
             await testUtils.form.clickEdit(form);
             await testUtils.dom.click(form.$('tbody td.o_field_x2many_list_row_add a'));
+            // use of owlCompatibilityNextTick because we have an x2many field with a boolean field
+            // (written in owl), so when we add a line, we sequentially render the list itself
+            // (including the boolean) field, so we have to wait for the next animation frame, and
```
Comment (Arcasias) [inline addons/web/static/tests/fields/relational_fields/field_one2many_tests.js:4557]:
```suggestion
            // (including the boolean field), so we have to wait for the next animation frame, and
```

File: addons/web/static/tests/fields/relational_fields/field_one2many_tests.js [hunk 3]
```diff
@@ -4571,6 +4588,12 @@ QUnit.module('fields', {}, function () {
                 },
             });
             await testUtils.dom.click(form.$('tbody td.o_field_x2many_list_row_add a'));
+            // use of owlCompatibilityNextTick because we have an x2many field with a boolean field
+            // (written in owl), so when we add a line, we sequentially render the list itself
+            // (including the boolean) field, so we have to wait for the next animation frame, and
```
Comment (Arcasias) [inline addons/web/static/tests/fields/relational_fields/field_one2many_tests.js:4593]:
```suggestion
            // (including the boolean field), so we have to wait for the next animation frame, and
```

File: addons/web/static/tests/fields/relational_fields/field_one2many_tests.js [hunk 4]
```diff
@@ -9331,6 +9372,74 @@ QUnit.module('fields', {}, function () {
 
             form.destroy();
         });
+
+        QUnit.test('mounted is called only once for x2many control panel', async function (assert) {
+            // This test could be remove as soon as the field widgets will be converted in owl.
```
Comment (Arcasias) [inline addons/web/static/tests/fields/relational_fields/field_one2many_tests.js:9377]:
```suggestion
            // This test could be removed as soon as the field widgets will be converted in owl.
```

File: addons/web/static/tests/helpers/test_utils.js [hunk 5]
```diff
@@ -97,6 +97,20 @@ odoo.define('web.test_utils', async function (require) {
         return testUtilsDom.returnAfterNextAnimationFrame();
     }
 
+    /**
+     * Calls nextTick. While we have an hybrid implemetation (Owl + legacy), we may
```
Comment (Arcasias) [inline addons/web/static/tests/helpers/test_utils.js:101]:
```suggestion
     * Calls nextTick. While we have a hybrid implemetation (Owl + legacy), we may
```

**Review by Arcasias — COMMENTED** [review]
In commit message:
- line 33: "thoses" -> "those"
- line 58: "their" -> "they"

**Comment by ged-odoo** [comment]
robodoo r+

**Comment by adr-odoo** [comment]
@robodoo r+

**Pull request closed without merging**